httpbin/core.py

```diff
@@ -649,6 +648,91 @@ def links(n):
     return redirect(url_for('link_page', n=n, offset=0))
 
 
+@app.route('/file/<path:p>', methods=('GET', 'POST',))
+def file(p):
+    """ File operations - create a new file, get a file or delete a file
+    :param p: directory path of the file for POST, full file path for GET and DELETE
+    :return:
+        GET: Returns a multi-part response with requested file(s)
+        POST: Returns the full path of the created file on success, error on error
+    """
+    if not args.file_endpoint:
+        return status_code(501) # Not Implemented
+
+    root_dir = os.path.join(app.config['SPOOL_DIR'], 'files')
+    if root_dir in ("", "/") or root_dir.startswith("/root"):
+        return status_code(500)
+
+    p = os.path.join(root_dir, p)
+    filename = request.args.get("name", None)
+
+    # for security
+    if '..' in p or (filename is not None and '..' in filename):
+        return status_code(404)
+
+    if request.method == 'GET':
+        ctime = request.args.get("ctime")
+        limit = request.args.get("limit")
+
+        bad_params=[]
+        if ctime is not None:
+            try:
+                ctime = int(ctime)
+            except ValueError:
+                bad_params.append('ctime')
+        if limit is not None:
+            try:
+                limit = int(limit)
+            except ValueError:
+                bad_params.append('limit')
+
+        if len(bad_params):
+            r = make_response("",400)
+            r.data = {"error": "url arguments {0} must be int".format(', '.join(bad_params))}
+            r.headers = {'Content-Type': 'application/json'}
+            return r
+
+        # add options:
+        ## grep for an expression in the file content
+
+        return return_files(dir_path=p, name=filename, ctime=ctime, limit=limit)
+
+    elif request.method == 'POST':
+        # A file of name 'filename' will be created in 'p' directory
+        # If the file already exists, create_file() will fail
+        data = request.data
+        ctype = request.headers.get('Content-Type', None)
+
+        res = create_file(path=p, content=data, filename=filename, ctype=ctype)
+        r = make_response()
+        if res:
+            r = status_code(201)
```

Comment:
you can make use of the httplib library for constants.  makes it easier to read and doesn't deal with magic numbers :)

https://docs.python.org/2/library/httplib.html

```python
import httplib
print httplib.OK
print httplib.INTERNAL_SERVER_ERROR
```